Library/Homebrew/cmd/tap.rb

```diff
@@ -42,59 +41,17 @@ def install_tap user, repo
     true
   end
 
-  def link_tap_formula(paths, warn_about_conflicts=true)
-    ignores = (HOMEBREW_LIBRARY/"Formula/.gitignore").read.split rescue []
-    tapped = 0
+  # migrate from symlink based tap structure to directories based structure
+  def migrate_taps(options={:autoskip => true})
+    ignore = HOMEBREW_LIBRARY/"Formula/.gitignore"
+    return if options[:autoskip] && !ignore.exist?
```

Comment:
Maybe I should phrase it this way: what are the different modes of operation of this method?